haystack/modeling/data_handler/data_silo.py

```diff
@@ -884,7 +884,7 @@ def _get_checksum(self):
             "max_seq_len": self.processor.max_seq_len,
             "dev_split": self.processor.dev_split,
             "tasks": self.processor.tasks,
-            "teacher_name_or_path": self.teacher.pipeline_config["params"]["model_name_or_path"],
+            "teacher_name_or_path": self.teacher.name,
```

Comment:
No it's not the same. `self.teacher` is a `FARMReader`, which inherits `name` from `BaseComponent`. While `pipeline_config` is also inherited from `BaseComponent`, its `model_name_or_path` has a different meaning than `name`. `name` is typically `Reader`, whereas `model_name_or_path` is typically `deepset/roberta-base-squad2` but it could also be a local file path. @ZanSara I believe this change needs to be reverted.